### ISSUE TITLE
version wrong?

### ISSUE THREAD

**Issue by xros** [zmap/zschema#2]
I looked the codes here:
https://github.com/zmap/zschema/blob/master/zschema/__init__.py

the version is 0.0.27

But the one on pypi is 0.0.32. 
https://pypi.python.org/pypi/zschema

I have installed through pypi.
> import zschema
    print(zschema.\__version\__)
    
It showed 0.0.27. But the site-packages is named of 0.0.32. So I think you might forget to change that version scope.

B.T.W.
How to use zschema to upload date into ElasticSearch? I haven't understood from [here](https://github.com/zmap/zschema)

Thanks!

**Comment by ppannuto** [comment]
@zakird or @dadrian this is an easy fix by having setup.py import the version from the library itself, e.g. https://github.com/ppannuto/python-titlecase/blob/master/setup.py#L10

**Comment by zakird** [comment]
@xros thanks for the issue. You're correct---our version numbers did not match up. We've moved some code around such that we can keep the two in sync. This will be fixed in 0.0.33.

**Comment by zakird** [comment]
In response to your other question, you can generate an Elastic Search schema with the following command:

```
zschema elasticsearch file.py:registered-schema-name
```

**Comment by xros** [comment]
@zakird thanks for you response. 

Can you tell me what  **registered-schema-name** means with **file.py**?

Do you mean that I have to write my own data schema in Python in the file **lib/python2.7/site-packages/zschema/registry.py** ?

Here I tried for several times but all failed

### Failed to generate ElasticSearch schema



#### Trial 1) Write my own Python schema script

* I wrote a file named **people.py** like this:
```python
from zschema.leaves import *
from zschema.compounds import *

people_schema = Record({
    "name":String(required=True),
    "street":String()
})
```
And then I ran it using this command
`zschema elasticsearch people.py:people_schema`

An error occurred.
> File "lib/python2.7/site-packages/zschema/registry.py", line │    __schemas[name] = schema
16, in get_schema                                                                                                                
    return __schemas[name]                                                                                                    
KeyError: 'people_schema'  


It looks I have to edit the file **lib/python2.7/site-packages/zschema/registry.py** to add an element in **__schemas[]**


#### Trial 2) Rewrite the registry.py  

* So I had added some lines of codes into that file **registry.py* and made it changed from
[This](https://github.com/zmap/zschema/blob/6ff068423e07fd7b917dafa8e983d281938c5af4/zschema/registry.py)

```python
__schemas = {}

def register_schema(name, schema):
    __schemas[name] = schema

def get_schema(name):
    return __schemas[name]
```



To this

```python
from zschema.leaves import *
from zschema.compounds import *

people_schema = Record({
"name":String(required=True),
"street":String()
})

__schemas = {"people_schema":people_schema}
#__schemas = {}

def register_schema(name, schema):
    __schemas[name] = schema

def get_schema(name):
    return __schemas[name]
```

Then I used this command (same as before) to get schema

```bash
zschema elasticsearch people.py:people_schema
```

And it returned something like this:
`{"people": {"properties": {"street": {"index": "not_analyzed", "type": "string"}, "name": {"index": "not_analyzed", "type": "string"}}}}`

It works but it does not returned a schema with defined data index and type. And I have to write Python schema in the **registry.py** file rather than the file **people.py** I wrote somewhere. 


#### Trial 3) Use any Python script to get ElasticSearch schema
* This time I used an empty Python script **empty_file.py** with the edited file **registry.py** which still looks like this

```python
from zschema.leaves import *
from zschema.compounds import *

people_schema = Record({
"name":String(required=True),
"street":String()
})

__schemas = {"people_schema":people_schema}
#__schemas = {}

def register_schema(name, schema):
    __schemas[name] = schema

def get_schema(name):
    return __schemas[name]
```

Then I used this command (same as before) to get schema

```bash
zschema elasticsearch empty_file.py:people_schema
```

And it returned something like this:
`{"people": {"properties": {"street": {"index": "not_analyzed", "type": "string"}, "name": {"index": "not_analyzed", "type": "string"}}}}`

**Same result as _Trial 2)_**


### Conclusions

1. Must write Python schema script in file **registry.py**.
2. Your self-written python schema file is useless.

### Suggestions

1. Make the self-written python scheme script work, please.
2. More documents about the data type mappings of elasticsearch, please!
3. Can we have an automatic script which can validate raw JSON file, generate its ElasticSearch schema and export data to ElasticSearch?


Many thanks!
Xros

**Comment by zakird** [comment]
1. You shouldn't need to edit anything or install anything in the source. You just need to call register_schema in your file. The name you pass in on the CLI is the registered name. Take a look at https://github.com/zmap/ztag/blob/master/ztag/schema.py for an example. So, conclusion (1) is incorrect.

2. Re conclusion (2): I don't know schema file is useless. We don't include any schema files with the project. 

3. Trial 2 looks like it returned the right output. That's a valid elasticsearch index. This are indexes that you can upload to elasticsearch by curling them. What are you expecting to be returned? 

4. If you want to see what types are matched to elasticsearch, these are all in https://github.com/zmap/zschema/blob/master/zschema/leaves.py, just look at ES_TYPE. 

5. ZSchema does not migrate data to elasticsearch. All it lets you do is generate a schema. If you want to autogenerate an ElasticSearch schema based on a file, then just upload the file to ElasticSearch---you don't know zschema. The intention of zschema is to generate a rigid schema and then validate data against that schema.

**Comment by xros** [comment]
@zakird Thanks for your help.

I know that zschema is focusing on validating data against defined schema. Yet I find zschema does nothing about uploading data to ElasticSearch. So I invented a wheel. Zschema is good at data structure scheming.

https://github.com/xros/json-py-es/    --  jsonpyes

I created this project to  load those JSON files ElasticSearch. And I think zschema would work fine with it.

Suggestions and improvements will be welcomed.
